Assets/Plugins/Source/Editor/Utility/GUIEditorUtility.cs

```diff
@@ -211,8 +213,480 @@ public static void AssigningEnumField<T>(string label, ref T value, float labelW
 
         private delegate T InputRenderDelegate<T>(string label, T value, float labelWidth, string tooltip);
 
-        public static List<string> RenderInputField(ConfigFieldAttribute configFieldDetails, List<string> value,
-            float labelWidth, string tooltip = null)
+        /// <summary>
+        /// Used to describe the function used to render a field of type T.
+        /// </summary>
+        /// <typeparam name="T">
+        /// The type being rendered.
+        /// </typeparam>
+        /// <param name="value">
+        /// The current value to display in the field.
+        /// </param>
+        /// <param name="options">
+        /// Optional parameters that describe the styling of the field to
+        /// render.
+        /// </param>
+        /// <returns>
+        /// The value entered.
+        /// </returns>
+        private delegate T RenderFieldDelegate<T>(T value, params GUILayoutOption[] options);
+
+        /// <summary>
+        /// Used to describe the function used to render a basic field of type
+        /// T.
+        /// </summary>
+        /// <typeparam name="T">
+        /// The type of input being rendered.
+        /// </typeparam>
+        /// <param name="rect">
+        /// The area in which the field should be rendered.
+        /// </param>
+        /// <param name="value">The value to put in the field.</param>
+        /// <returns>The value entered in the field.</returns>
+        private delegate T RenderBasicFieldDelegate<T>(Rect rect, T value);
```

Comment:
Does "basic" imply "does not contain GUILayoutOptions and instead has Rect"? What makes something "Basic"?

Reply:
It's a bit of a historical thing. Unity's editor UI toolkit is built upon ImGUI and has undergone several refactors such that Unity actually has a few different ways to render things in ImGUI. The _newer_ method does not require that you specify the `Rect` to draw in, but the _older_ (hence I'm calling "basic") method _does_. 

The control that I use to render lists requires use of the old set of methods instead of the new ones - so when the delegate needs to be used in that context, it will have different parameters.